File: 3ra Practica de 121/GENERICIDAD/3-catalogo/codigo/Catalogo.py
class Catalogo:
    def __init__(self):
        self.elementos = []

    def agregar(self, elemento):
        self.elementos.append(elemento)

    def buscar(self, indice):
        if 0 <= indice < len(self.elementos):
            return self.elementos[indice]
        raise IndexError("Índice fuera de rango")

if __name__ == "__main__":
    catalogo_libros = Catalogo()
    catalogo_libros.agregar("Cien años de soledad")
    catalogo_libros.agregar("1984")
    print("Libro en posición 0:", catalogo_libros.buscar(0))

    catalogo_productos = Catalogo()
    catalogo_productos.agregar(100)
    catalogo_productos.agregar(200)
    print("Producto en posición 1:", catalogo_productos.buscar(1))

File: 3ra Practica de 121/GENERICIDAD/3-catalogo/codigo/Catalogo.java
import java.util.ArrayList;
import java.util.List;

public class Catalogo<T> {
    private List<T> elementos;

    public Catalogo() {
        this.elementos = new ArrayList<>();
    }

    public void agregar(T elemento) {
        elementos.add(elemento);
    }

    public T buscar(int indice) {
        if (indice >= 0 && indice < elementos.size()) {
            return elementos.get(indice);
        }
        throw new IndexOutOfBoundsException("Índice fuera de rango");
    }

    public static void main(String[] args) {
        Catalogo<String> catalogoLibros = new Catalogo<>();
        catalogoLibros.agregar("Cien años de soledad");
        catalogoLibros.agregar("1984");
        System.out.println("Libro en posición 0: " + catalogoLibros.buscar(0));

        Catalogo<Integer> catalogoProductos = new Catalogo<>();
        catalogoProductos.agregar(100);
        catalogoProductos.agregar(200);
        System.out.println("Producto en posición 1: " + catalogoProductos.buscar(1));
    }
}